homeassistant/components/ambient_station/sensor.py

```diff
@@ -1,20 +1,557 @@
 """Support for Ambient Weather Station sensors."""
 from __future__ import annotations
 
-from homeassistant.components.sensor import DOMAIN as SENSOR, SensorEntity
+from homeassistant.components.sensor import SensorEntity, SensorEntityDescription
 from homeassistant.config_entries import ConfigEntry
-from homeassistant.const import ATTR_NAME
+from homeassistant.const import (
+    ATTR_NAME,
+    CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
+    CONCENTRATION_PARTS_PER_MILLION,
+    DEGREE,
+    DEVICE_CLASS_CO2,
+    DEVICE_CLASS_HUMIDITY,
+    DEVICE_CLASS_ILLUMINANCE,
+    DEVICE_CLASS_PM25,
+    DEVICE_CLASS_PRESSURE,
+    DEVICE_CLASS_TEMPERATURE,
+    DEVICE_CLASS_TIMESTAMP,
+    IRRADIATION_WATTS_PER_SQUARE_METER,
+    LIGHT_LUX,
+    PERCENTAGE,
+    PRECIPITATION_INCHES,
+    PRECIPITATION_INCHES_PER_HOUR,
+    PRESSURE_INHG,
+    SPEED_MILES_PER_HOUR,
+    TEMP_FAHRENHEIT,
+)
 from homeassistant.core import HomeAssistant, callback
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
 
-from . import (
-    SENSOR_TYPES,
-    TYPE_SOLARRADIATION,
-    TYPE_SOLARRADIATION_LX,
-    AmbientStation,
-    AmbientWeatherEntity,
+from . import TYPE_SOLARRADIATION, TYPE_SOLARRADIATION_LX, AmbientWeatherEntity
+from .const import (
+    ATTR_LAST_DATA,
+    DATA_CLIENT,
+    DOMAIN,
+    TYPE_24HOURRAININ,
+    TYPE_BAROMABSIN,
+    TYPE_BAROMRELIN,
+    TYPE_CO2,
+    TYPE_DAILYRAININ,
+    TYPE_DEWPOINT,
+    TYPE_EVENTRAININ,
+    TYPE_FEELSLIKE,
+    TYPE_HOURLYRAININ,
+    TYPE_HUMIDITY,
+    TYPE_HUMIDITY1,
+    TYPE_HUMIDITY2,
+    TYPE_HUMIDITY3,
+    TYPE_HUMIDITY4,
+    TYPE_HUMIDITY5,
+    TYPE_HUMIDITY6,
+    TYPE_HUMIDITY7,
+    TYPE_HUMIDITY8,
+    TYPE_HUMIDITY9,
+    TYPE_HUMIDITY10,
+    TYPE_HUMIDITYIN,
+    TYPE_LASTRAIN,
+    TYPE_MAXDAILYGUST,
+    TYPE_MONTHLYRAININ,
+    TYPE_PM25,
+    TYPE_PM25_24H,
+    TYPE_PM25_IN,
+    TYPE_PM25_IN_24H,
+    TYPE_SOILHUM1,
+    TYPE_SOILHUM2,
+    TYPE_SOILHUM3,
+    TYPE_SOILHUM4,
+    TYPE_SOILHUM5,
+    TYPE_SOILHUM6,
+    TYPE_SOILHUM7,
+    TYPE_SOILHUM8,
+    TYPE_SOILHUM9,
+    TYPE_SOILHUM10,
+    TYPE_SOILTEMP1F,
+    TYPE_SOILTEMP2F,
+    TYPE_SOILTEMP3F,
+    TYPE_SOILTEMP4F,
+    TYPE_SOILTEMP5F,
+    TYPE_SOILTEMP6F,
+    TYPE_SOILTEMP7F,
+    TYPE_SOILTEMP8F,
+    TYPE_SOILTEMP9F,
+    TYPE_SOILTEMP10F,
+    TYPE_TEMP1F,
+    TYPE_TEMP2F,
+    TYPE_TEMP3F,
+    TYPE_TEMP4F,
+    TYPE_TEMP5F,
+    TYPE_TEMP6F,
+    TYPE_TEMP7F,
+    TYPE_TEMP8F,
+    TYPE_TEMP9F,
+    TYPE_TEMP10F,
+    TYPE_TEMPF,
+    TYPE_TEMPINF,
+    TYPE_TOTALRAININ,
+    TYPE_UV,
+    TYPE_WEEKLYRAININ,
+    TYPE_WINDDIR,
+    TYPE_WINDDIR_AVG2M,
+    TYPE_WINDDIR_AVG10M,
+    TYPE_WINDGUSTDIR,
+    TYPE_WINDGUSTMPH,
+    TYPE_WINDSPDMPH_AVG2M,
+    TYPE_WINDSPDMPH_AVG10M,
+    TYPE_WINDSPEEDMPH,
+    TYPE_YEARLYRAININ,
+)
+
+SENSOR_DESCRIPTIONS = (
+    SensorEntityDescription(
+        key=TYPE_24HOURRAININ,
```

Comment:
I'm wondering if it's really necessary to define constants for all keys. As best as I can tell, they aren't used anywhere else.

Same for the binary sensors, although some are used in the lookup.

Reply:
There are enough instances where those constants are used elsewhere that I don't want to mix/match (i.e., some strings are constants, some are defined in-place).